src/interval.jl

```diff
@@ -469,6 +714,100 @@ function Base.merge(a::AbstractInterval, b::AbstractInterval)
     return Interval(left, right)
 end
 
+Base.union(x::AbstractInterval) = x
+
+const AbstractIntervals = Union{AbstractInterval, AbstractVector{<:AbstractInterval}}
+mergeclean(x, y) = unbunch(union(x), union(y))
+function Base.intersect(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx && iny, mergeclean(x, y)...)
+end
+function Base.union(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx || iny, mergeclean(x, y)...)
+end
+function Base.setdiff(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx && !iny, mergeclean(x, y)...)
+end
+function Base.symdiff(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx ⊻ iny, mergeclean(x, y)...)
+end
+function Base.issubset(x::AbstractIntervals, y::AbstractIntervals)
+    return isempty(setdiff(x, y))
+end
+# may or may not be from Base (see top of `Intervals.jl`)
+function isdisjoint(x::AbstractIntervals, y::AbstractIntervals)
+    return isempty(intersect(x, y))
+end
+Base.in(x, y::AbstractVector{<:AbstractInterval}) = any(yᵢ -> x ∈ yᵢ, y)
+function Base.issetequal(x::AbstractIntervals, y::AbstractIntervals)
+    x, y = mergeclean(x,y)
+    return x == y || (all(isempty, bunch(x)) && all(isempty, bunch(y)))
+end
+
+Base.length(x::AbstractInterval) = 1
+
+# sort endpoints so that a closed left endpoint comes before
+# a closed right endpoint (used by `find_intersections`)
+struct EndpointOffset{E <: Endpoint}
+    data::E
+end
+offset(x::Endpoint) = EndpointOffset(x)
+function Base.isless(x::EndpointOffset, y::EndpointOffset)
+    if isequal(x.data, y.data)
+        return isless(offset_value(x.data), offset_value(y.data))
+    else
+        return isless(x.data, y.data)
+    end
+end
+
+"""
+    find_intersections(x::Union{AbstractInterval, AbstractVector{<:AbstractInterval}}, 
+                       y::Union{AbstractInterval, AbstractVector{<:AbstractInterval}}; sorted=false)
+
+Returns a Vector{Vector{Int}} where the value at index i gives the indices to
+all intervals in `y` that intersect with `x[i]`.
+
+"""
+function find_intersections(x_::AbstractIntervals, y_::AbstractIntervals)
+    xa = vcat(x_)
+    x = unbunch(xa; enumerate=true, sortby=offset)
+    y = unbunch(vcat(y_); enumerate=true, sortby=offset)
+    result = [Vector{Int}() for _ in 1:length(xa)]
+
+    find_intersections_helper(result, x, y)
```

Comment:
```suggestion
    result = [Vector{Int}() for _ in 1:length(xa)]

    find_intersections_helper(result, x, y)
```
Function is mutating and should marked as such